src/runtime/CompressibleString.cpp

```diff
@@ -217,13 +217,22 @@ bool CompressibleString::compressWorker(void* callerSP)
         m_compressedData.push_back(std::vector<char>(compBuffer.get(), compBuffer.get() + compressedLength));
     }
 
-    m_context->vmInstance()->compressibleStringsUncomressedBufferSize() -= decomressedBufferSize();
+    m_vmInstance->compressibleStringsUncomressedBufferSize() -= decomressedBufferSize();
 
     // immediately free the original string after compression when there is no reference on stack
-    deallocateStringDataBuffer(const_cast<void*>(m_bufferData.buffer));
+    deallocateStringDataBuffer(const_cast<void*>(m_bufferData.buffer), m_bufferData.length * (m_bufferData.has8BitContent ? 1 : 2));
 
     m_bufferData.bufferAs8Bit = nullptr;
     m_isCompressed = true;
+
+    /*
+    size_t compressedSize = 0;
+    for (size_t i = 0; i < m_compressedData.size(); i ++) {
+        compressedSize += m_compressedData[i].size();
+    }
+    ESCARGOT_LOG_INFO("CompressibleString::compressWorker %fKB -> %fKB\n", originByteLength / 1024.f, compressedSize / 1024.f);
+    */
```

Comment:
This comment needs to be removed if its no longer necessary.

Reply:
these comment is used on memory optimization